src/tests/functions/Add-DscResourceModule.Tests.ps1

```diff
@@ -0,0 +1,63 @@
+Describe 'Vendoring a DSC module' {
+  InModuleScope puppet.dsc {
+    Context 'When only -Name is specified' {
+      Mock New-Item -Verifiable
+      Mock Save-Module -Verifiable
+      Mock Test-Path { $false } -ParameterFilter { $path -eq './tmp' }
+      Mock Get-CanonicalPath { './tmp' }
+      Mock Test-Path { $false } -ParameterFilter { $path -eq './tmp_tmp' }
+      Mock Get-ChildItem
+      Mock Move-Item
+      Mock Remove-Item
+
+      Add-DscResourceModule -Name SomeDSCModule
+
+      It 'downloads the latest version of the module to ./tmp' {
+        Assert-VerifiableMock
+        Assert-MockCalled New-Item -Times 1 -ParameterFilter { ($path -eq './tmp') -and ($ItemType -eq 'Directory') }
+        Assert-MockCalled New-Item -Times 1 -ParameterFilter { ($path -eq './tmp_tmp') -and ($ItemType -eq 'Directory') }
+        Assert-MockCalled Save-Module -Times 1 -Debug -ParameterFilter {
+          ($Name -eq 'SomeDSCModule') `
+            -and ($path -eq './tmp_tmp') `
+            -and ($RequiredVersion -eq '')
+        }
+        Assert-MockCalled Remove-Item -Times 1 -ParameterFilter { $path -eq './tmp_tmp' }
+      }
+
+      It 'does not leave litter behind' {
+        Get-ChildItem . -filter tmp | Should -BeNullOrEmpty
```

Comment:
Can this test fail if we mock `New-Item`?

Reply:
I've had troubles during tinkering around with the code that directories were created by the tests. I'll revisit whether this is still needed

Reply:
We may also be able to isolate any actual creations into the [test drive Pester exposes](https://pester.dev/docs/usage/testdrive).

Reply:
using the testdrive now.